test/Analysis/firrtl-test-instance-info.mlir

```diff
@@ -0,0 +1,59 @@
+// RUN: circt-opt %s -test-firrtl-instance-info 2>&1 | FileCheck %s
+
+firrtl.circuit "Foo" {
+  firrtl.layer @A bind {
+  }
+  // CHECK:      firrtl.module @Corge
+  // CHECK-NEXT:   isDut: false
+  // CHECK-NEXT:   isUnderDut: false
+  // CHECK-NEXT:   isFullyUnderDut: false
+  // CHECK-NEXT:   isUnderLayer: true
+  // CHECK-NEXT:   isFullyUnderLayer: false
+  firrtl.module @Corge() {}
```

Comment:
Consider making most of these modules private.  Once we land/start using multiple public modules, I'd guess this pass will want to walk down from all entry points not only a single one?

That said I know you prefer minimal tests so if you'd rather update it later (when adding support) that works for me too 👍 .

Reply:
Good suggestion. Changed in 706c9752b. I made the DUT private, too, but this may be wrong.

The lack of `private` was more motivated by simplifying the FileCheck check lines. I'll switch those to be slightly terser and match on only the symbol and avoid the regex the ignore private.